testing/scenario_app/ios/Scenarios/ScenariosUITests/LocalizationInitializationTest.m

```diff
@@ -0,0 +1,59 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "LocalizationInitialization.h"
+
+FLUTTER_ASSERT_ARC
+
+@interface XCUIElement (ftr_waitForNonExistence)
+/// Keeps waiting until the element doesn't exist.  Returns NO if the timeout is
+/// reached before it doesn't exist.
+- (BOOL)ftr_waitForNonExistenceWithTimeout:(NSTimeInterval)timeout;
+/// Waits the full duration to ensure something doesn't exist for that duration.
+/// Returns NO if at some point the element exists during the duration.
+- (BOOL)ftr_waitForNonExistenceForDuration:(NSTimeInterval)duration;
+@end
+
+@implementation XCUIElement (ftr_waitForNonExistence)
+- (BOOL)ftr_waitForNonExistenceWithTimeout:(NSTimeInterval)timeout {
+  NSTimeInterval delta = 0.5;
+  while (timeout > 0.0) {
+    if (!self.exists) {
+      return YES;
+    }
+    usleep(delta * 1000000);
```

Comment:
Can we avoid this? 

Would it help at all if we had EarlyGrey for Flutter instead?

If we need that, we should probably add a TODO here to replace this when EarlGrey is available. 
/cc @jmagman who might know a good trick for this

Reply:
Something like:
```obj-c
NSPredicate *predicate = [NSPredicate predicateWithFormat:@"exists = YES"];
XCTestExpectation *expectation = [self expectationForPredicate:predicate evaluatedWithObject:textInputSemanticsObject handler:NULL];
[self waitForExpectations:@[expectation] timeout:10.0];
```

Reply:
This is probably better, actually:
```obj-c
XCTestExpectation *expectation = [self keyValueObservingExpectationForObject:textInputSemanticsObject keyPath:@"exists" expectedValue:@YES];
[self waitForExpectations:@[expectation] timeout:10.0];
```
(I didn't actually test these).

Reply:
It's coming back to me now - that week I got better at Objective C with your guidance .. and how it faded...

Reply:
I had to look it up myself, it's been awhile...

Reply:
Ahh thanks for the assistance, Trying to learn all the iOS testing frameworks and whatnot to write this haha. This particular code was repurposed from another test, still working on refining/improving it.